File: hola-mundo-backend-2022-grupo-7/src/main/kotlin/trabajoPracticoAlgoritmos2/Puntuacion.kt
package trabajoPracticoAlgoritmos2

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

class Puntuacion (
    val numero:Int,
    @JsonIgnore
    val creador: Usuario?
    ){

    @JsonProperty("creador")
    fun getCreatorAsString(): UsuarioDTO? {
        return creador?.let { UsuarioDTO(it.id,creador.nombre, creador.apellido, creador.username,creador.paisDeResidencia) }
    }

    fun validacion():Boolean{
        if(numero < 1 || numero > 10){
            return false
        }
        return true
    }
}

File: Angular/src/Dominio/Puntuaciones/Puntuacion.ts
import { Usuario } from "../Usuario/Usuario"

export class Puntuacion{
 
    readonly numero:number
    readonly creador:Usuario

    constructor(_valor:number,_creador:Usuario){
        this.numero = _valor
        this.creador = _creador
    }

    validacion():boolean{
        return !(this.numero < 1 || this.numero > 10)
    }
}